### ISSUE TITLE
[Bug report] Learning rate of BertAdam may become negative

### ISSUE THREAD

**Issue by HuXiangkun** [huggingface/pytorch-pretrained-BERT#132]
When using 'warmup_linear', step could be larger than t_total, then lr becomes negative.

I found this bug in run_squad.py, which using number of training examples as total step, but number of training features is reasonable.

**Comment by HuXiangkun** [comment]
NONE
